sync/src/relayer/compact_block_process.rs

```diff
@@ -234,21 +237,18 @@ impl<'a, CS: ChainStore + 'static> CompactBlockProcess<'a, CS> {
         }
 
         let fbb = &mut FlatBufferBuilder::new();
-        let message = RelayMessage::build_get_block_transactions(
-            fbb,
-            &block_hash,
-            &missing_indexes
-                .into_iter()
-                .map(|i| i as u32)
-                .collect::<Vec<_>>(),
-        );
+        let message = RelayMessage::build_get_block_transactions(fbb, &block_hash, &indexes);
         fbb.finish(message, None);
         if let Err(err) = self
             .nc
             .send_message_to(self.peer, fbb.finished_data().into())
         {
             ckb_logger::debug!("relayer send get_block_transactions error: {:?}", err);
         }
+
+        let mut pending_get_block_transactions =
+            self.relayer.shared().pending_get_block_transactions();
+        pending_get_block_transactions.insert((block_hash, self.peer), indexes);
```

Comment:
As for a CompactBlock with fresh transactions, here we record the CompactBlock inside `pending_compact_blocks`, and record the fresh transaction-ids inside `pending_get_block_transactions`. I think it is redundant and hard to keep consistency between 2 caches. 

So, 1. does it neccesary to cache `pending_get_block_transactions`? 2. If it is neccesary, can we merge `pending_get_block_transactions` and `pending_compact_blocks` into one?

Reply:
Thanks, updated, pls review again.